src/metrics/perf_event.rs

```diff
@@ -80,12 +80,8 @@ impl HardwareMetricGroup {
     ) -> std::io::Result<()> {
         let iterator = perf_events.counters.iter();
         for value in iterator {
-            register_histogram!(self.get_field_gauge_name(&node_name, &value.to_string()));
+            register_histogram!(value.to_string(),"node" => String::from (node_name.clone()));
         }
         self.group.enable()
```

Comment:
ah didn't catch this one before. Why is the group enabled here?

But could we remove the whole function? And move the registration out of the struct and handle it in the Node.

```rust
// in Node::new()
for counter in hardware_metric_group.counters() {
  register_histogram!(counter, "node" => self.descriptor.clone());
}
```